shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/init/ContextRegisterListener.java

```diff
@@ -28,35 +28,36 @@
 import org.apache.shenyu.register.common.dto.URIRegisterDTO;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.boot.web.context.WebServerInitializedEvent;
 import org.springframework.context.ApplicationListener;
-import org.springframework.context.event.ContextRefreshedEvent;
 import org.springframework.core.env.Environment;
 import org.springframework.lang.NonNull;
 
+import java.util.Optional;
 import java.util.Properties;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * The type Context register listener.
  */
-public class ContextRegisterListener implements ApplicationListener<ContextRefreshedEvent> {
-    
+public class ContextRegisterListener implements ApplicationListener<WebServerInitializedEvent> {
```

Comment:
WebServerInitializedEvent  it is spring version？， if user is spring 4.x，Does that not work?

Reply:
when I use springboot 1.5.22.RELEASE and shenyu-spring-boot-starter-client-springcloud 2.4.2, springboot can not start.

Reply:
I guess `WebServerInitializedEvent` can not work.

Reply:
Think it can be implemented in combination with @ConditionalOnClass, I will reimplement it later.